### ISSUE TITLE
Mac support

### ISSUE THREAD

**Issue by steshaw** [haskell-CI/haskell-ci#55]
It would be nice to have Mac support since Travis supports it. I wonder if anyone else is interested in it?

The Idris project is an example of [what it takes to make a multi-gcc-travis .travis.yml file into on which supports Mac](https://github.com/idris-lang/Idris-dev/blob/master/.travis.yml).

I'd be happy to work on a patch.

**Comment by phadej** [comment]
FWIW: cabal tests on mac osx too: https://github.com/haskell/cabal/blob/master/travis-install.sh

IMHO using brew to install ghc is problematic, when we want particular version of GHC. There are official bindist for `cabal-install` as well.

**Comment by steshaw** [comment]
@phadej right, I've got no problem with the idea of not using Homebrew to grab `ghc` and `cabal-install`. That `travis-install.sh` looks just the kind of thing to crib from!

**Comment by hvr** [comment]
@steshaw I've been wondering how to add OSX support, and this was mainly blocked by not having the equivalent of ppa packages for OSX. Otoh, doing it like cabal does seems interesting. Should be fairly easy to combine with the `new-build` script generator. If we have the ghc bindists, we only need an executable from a recent cabal-1.24 branch snapshot (unfortunately, cabal-1.25 is currently slightly broken) and possibly happy+alex executables, and we'd be ready to go.

**Comment by steshaw** [comment]
@hvr, what do you mean by "doing it like cabal"?

The idris project only uses cabal-1.24 with ghc-8.0.1. I guess you need cabal 1.24 to get the nix-style `new-build` enhancement? Also, I thought I noticed in `make_travis_yml_2.hs` that there's no support for `cabal test` with the new nix-style builds.

At least for `idris`, this means that we'd be using `make_travis_yml.hs` for a bit.

**Comment by phadej** [comment]
There's now `--osx` flag. Please test it out!
